server/src/main/java/io/crate/fdw/ServersMetadata.java

```diff
@@ -339,6 +360,17 @@ public Iterable<UserMapping> getUserMappings() {
                 ).iterator();
     }
 
+    public Iterable<UserMappingOptionsTableInfo.UserMappingOptions> getUserMappingOptions() {
+        return () ->
+            servers.values().stream()
+                .map(server -> server)
+                .flatMap(server -> server.users.entrySet().stream()
+                    .flatMap(e -> e.getValue().getAsStructuredMap().entrySet().stream()
```

Comment:
This should probably end up using the `getAsStructuredMap` overload that takes a set of masked settings. And the masked settings need to be provided from the concrete fdw implementation.

Maybe easier to split this PR into two parts: First setting validation, and then follow up with the information schema table.

Reply:
Could we instead mask all option values(not only password) without proper accesses like how Postgres handles it?

https://www.postgresql.org/docs/current/infoschema-user-mapping-options.html
>  This column will show as null unless the current user is the user being mapped, or the mapping is for PUBLIC and the current user is the server owner, or the current user is a superuser

Reply:
Not sure. I'd tend to use the masking system we have in place, to keep it consistent with sys.repositories.
The username could be visible for all and the password masked. It would also reduce the amount of custom logic required in https://github.com/crate/crate/pull/15591/files#diff-0d97afa2e4b2cf61789649f72feb9a086b2b0489f5a4a61229a53890a8deaa6cR133

Let's see what the others think.

@seut @matriv

Reply:
Actually, let's go ahead with this and get this in - we can still change it afterwards based on the feedback